Packet++/src/PacketUtils.cpp

```diff
@@ -49,14 +55,16 @@ uint16_t computeChecksum(ScalarBuffer<uint16_t> vec[], size_t vecSize)
 	{
 		sum = (sum & 0xffff) + (sum >> 16);
 	}
-
 	PCPP_LOG_DEBUG("Sum before invert = " << sum << ", 0x" << std::uppercase << std::hex << sum);
 
-	sum = ~sum;
+	// To obtain the checksum we take the ones' complement of this result
+	uint16_t result = sum;
+	result = ~result;
 
-	PCPP_LOG_DEBUG("Calculated checksum = " << sum << ", 0x" << std::uppercase << std::hex << sum);
+	PCPP_LOG_DEBUG("Calculated checksum = " << sum << ", 0x" << std::uppercase << std::hex << result);
 
-	return ((uint16_t) sum);
+	// We return the result in Network byte order
+	return htons(result);

```

Comment:
not obvious in the first place - but this is a nicer way than doing it on temp sums (line 43).

Reply:
maybe i'm wrong but i think the previous way was wrong as then the carry count would be wrong.

Reply:
I think it was also OK. because of https://datatracker.ietf.org/doc/html/rfc1071 Page 3, (B)  Byte Order Independence

Reply:
btw. why do we do the carry twice? once on the local sum and once on sum. Should be enough to do it only on sum.

please have a look at the original version of tcpdump. I very much like the simplicity of it:
https://github.com/the-tcpdump-group/tcpdump/blob/a91d20cc8491f2afcc04e51474244710481244b9/print-ip.c#L276

also the implementation and comment of the next function is notable:
https://github.com/the-tcpdump-group/tcpdump/blob/a91d20cc8491f2afcc04e51474244710481244b9/print-ip.c#L312